logdna/logdna.py

```diff
@@ -100,27 +103,27 @@ def flush(self):
                         'tags': self.tags if self.tags else None},
                     stream=True,
                     timeout=self.request_timeout)
+                res.raise_for_status()
+                # when no RequestException happened
+                self.exception_flag = False
                 self.buf = []
                 self.buf_byte_length = 0
                 if self.flusher:
                     self.flusher.cancel()
                     self.flusher = None
-                self.lock.release()
                 # Ensure messages that could've dropped are appended back onto buf
                 self.buf = self.buf + self.secondary
                 self.secondary = []
+                self.lock.release()
         except requests.exceptions.RequestException as e:
```

Comment:
Should we consider all exceptions or just Network-related ones?

Reply:
It will catch all unsuccessful responses from the server.